crates/bevy_ecs/src/system/query.rs

```diff
@@ -1194,7 +1171,10 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> Query<'w, 's, Q, F> {
     #[inline]
     #[track_caller]
     pub fn component_mut<T: Component>(&mut self, entity: Entity) -> Mut<'_, T> {
-        match self.get_component_mut(entity) {
+        // SAFETY: unique access to query (preventing aliased access)
+        let result = unsafe { self.get_component_unchecked_mut(entity) };
+
+        match result {
```

Comment:
Hold up -- why was this changed to use `get_component_unchecked_mut`? Can we go back to using the safe method?

Reply:
Yeah that makes more sense. I think at the time I thought it made more sense to directly access the `QueryState` version of that function. Fixed now!